packages/ui/src/models/camel-entities/flows/kamelet-binding.ts

```diff
@@ -2,11 +2,16 @@ import { KameletBinding as KameletBindingModel } from '@kaoto-next/camel-catalog
 import { v4 as uuidv4 } from 'uuid';
 import { VisualizationNode } from '../../visualization';
 import { BaseVisualCamelEntity, EntityType } from '../base-entity';
-import { KameletBindingSink, KameletBindingSource, KameletBindingStep, KameletBindingSteps } from '../kamelet-binding';
+import {
+  KameletBindingSink,
+  KameletBindingSource,
+  KameletBindingStep,
+  KameletBindingSteps,
+} from '../kamelet-binding-overrides';
 
 export class KameletBinding implements BaseVisualCamelEntity {
-  readonly id = uuidv4();
-  readonly type = EntityType.KameletBinding;
+  id = uuidv4();
```

Comment:
[curious-question] Do we need to remove the `readonly` modifier?

Reply:
with `readonly` for property `type`, there was following error in the Pipe class :

Property 'type' in type 'Pipe' is not assignable to the same property in base type 'KameletBinding'.

Reply:
Oh, interesting... No worries, I'll take a look later then 👍